ci-operator/step-registry/ovn/sdn-migration-rollback/ovn-sdn-migration-rollback-commands.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+set -x
+set -o errexit
+set -o nounset
+set -o pipefail
+
+TARGET=${TARGET:-OpenShiftSDN}
+
+oc patch MachineConfigPool master --type='merge' --patch '{"spec":{"paused":true}}'
+oc patch MachineConfigPool worker --type='merge' --patch '{"spec":{"paused":true}}'
+
+# Reset the spec.migration before we can set it to other value
+oc patch Network.operator.openshift.io cluster --type='merge' --patch '{"spec":{"migration":null}}'
+# Wait until CNO update the applied-cluster cm to the latest
+timeout 60s bash <<EOT
+until 
+  ! oc get network -o yaml | grep migration  > /dev/null
+do
+  echo "migration field is not cleaned by CNO"
+  sleep 3
+done
+EOT
+
+# Allow rollback
+# Change network to target network in Network.config.openshift.io the CR to trigger machine config update by MCO.
+oc patch Network.operator.openshift.io cluster --type='merge' --patch "{\"spec\":{\"migration\":{\"networkType\":\"${TARGET}\"}}}"
+oc patch Network.config.openshift.io cluster --type='merge' --patch "{\"spec\":{\"networkType\":\"${TARGET}\"}}"
+
+oc wait co network --for='condition=PROGRESSING=True' --timeout=30s
+# Wait until the multus pods are restarted
+timeout 300 oc rollout status ds/multus -n openshift-multus
+
+# Reboot all the nodes
+
+oc get pod -n openshift-machine-config-operator | grep daemon|awk '{print $1}'|xargs -i oc rsh -n openshift-machine-config-operator {} chroot /rootfs shutdown -r +1
+
+# Wait until all nodes reboot and the api-server is unreachable.
+timeout 120s bash <<EOT
+until
+  ! oc get node > /dev/null;
+do
+  echo "nodes not rebooted"
+  sleep 10
+done
+EOT
+
+# Wait for nodes come back
+timeout 1800s bash <<EOT
+until
+  oc wait node --all --for condition=ready --timeout=10s;
+do
+  echo "nodes not ready"
+  sleep 10
+done
+EOT
+
+# Resume MCPs after reboot
+timeout 1800s bash <<EOT
+until
+  oc patch MachineConfigPool master --type='merge' --patch '{"spec":{"paused":false}}' && \
+  oc patch MachineConfigPool worker --type='merge' --patch '{"spec":{"paused":false}}';
+do
+  sleep 10;
+done
+EOT
+
+oc wait mcp --all --for='condition=UPDATING=True' --timeout=300s
+
+# Wait until MCO finishes its work or it reaches the 20mins timeout
+timeout 1200s bash <<EOT
+until
+  oc wait mcp --all --for='condition=UPDATED=True' --timeout=10s && \
+  oc wait mcp --all --for='condition=UPDATING=False' --timeout=10s && \
+  oc wait mcp --all --for='condition=DEGRADED=False' --timeout=10s; 
+do
+  sleep 10
+  echo "Some MachineConfigPool Degraded=True,Progressing=True,or Available=False";
+done
+EOT
+
+# Check all cluster operators back to normal
```

Comment:
Perhaps we should be checking CVO status? it should encapsulate all operator status

Reply:
By checking the ClusterOperater, we can find out which operator blocks the migration, if anything goes wrong.

Reply:
We'll get more info from `gather-extra` step anyway, but okay, lets leave the script in place in case info cannot be collect afterwards